.github/workflows/push.yml

```diff
@@ -72,69 +71,103 @@ jobs:
           echo "::set-output name=is_semantic_tag::$IS_SEMANTIC_TAG"
           echo "::set-output name=version::$VERSION"
 
-  build_image:
+  test:
     needs:
       - set_version
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - name: linux
+            os: ubuntu-latest
+            projects: TEST_KBC_PROJECTS_LINUX
+            experimental: false
+          - name: mac-os
+            os: macos-latest
+            projects: TEST_KBC_PROJECTS_MACOS
+            experimental: false
+          - name: windows
+            os: windows-latest
+            projects: TEST_KBC_PROJECTS_WINDOWS
+            experimental: true
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
+    name: test ${{ matrix.name }}
     steps:
-      - name: Check out the repo
+      - name: Setup line endings
+        run: git config --global core.autocrlf false
+      - name: Checkout code
         uses: actions/checkout@v2
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
+      - name: Install Go
+        uses: actions/setup-go@v2
         with:
-          username: ${{ env.DOCKERHUB_USER }}
-          password: ${{ env.DOCKERHUB_TOKEN }}
-      - name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
+          go-version: ${{ env.GO_VERSION }}
+      - name: Install tools
+        run: . ./scripts/tools.sh
+        shell: bash
+      - name: Load Go cache
+        uses: ./.github/actions/go-cache
         with:
-          push: true
-          tags: ${{ env.IMAGE }}:${{ env.IMAGE_TAG }}
-          cache-from: type=registry,ref=${{ env.IMAGE }}:buildcache
-          cache-to: type=registry,ref=${{ env.IMAGE }}:buildcache,mode=max
+          key: test
+      - name: Run tests
+        run: |
+          make mod
+          make tests
+        shell: bash
+        env:
+          VERSION: ${{ needs.set_version.outputs.version }}
+          TEST_KBC_PROJECTS: ${{ env[matrix.projects] }}
+          TEST_PARALLELISM: 16
```

Comment:
Testovanim som zistil, ze najrychlesie to zbehne, ked sa spusta `16` testov paralelne.